target/ppc/translate/fixedpoint-impl.c.inc

```diff
@@ -139,6 +227,38 @@ TRANS64(STDU, do_ldst_D, true, true, MO_Q)
 TRANS64(STDUX, do_ldst_X, true, true, MO_Q)
 TRANS64(PSTD, do_ldst_PLS_D, false, true, MO_Q)
 
+/* Store Quadword */
+static bool trans_STQ(DisasContext *ctx, arg_D *a)
+{
+    REQUIRE_64BIT(ctx);
+
+    if (unlikely(a->rt & 1)) {
+        gen_inval_exception(ctx, POWERPC_EXCP_INVAL_INVAL);
+        return true;
+    }
+
+    if (!(ctx->insns_flags2 & PPC2_LSQ_ISA207)) {
+        if (ctx->pr) {
```

Comment:
I don't know when/if ctx->pr is updated. Checking the code, I couldn't see it being updated anywhere, so unless it is some hidden thing of mmu code, I think it's better to use msr_pr instead of ctx->pr; Same goes for the trans_LQ

@alqotel do you know anything I don't?

Reply:
It's set in `ppc_tr_init_disas_context`, like `ctx->altivec_enabled` and `ctx->vsx_enabled`. I'll look further in differences between `ctx->pr` and `msr_pr`, but all code in `translate.c` uses `ctx->pr`, including the original implementation of lq/stq.

Reply:
I see.... then probably ctx->pr is the correct thing. msr_pr uses the MSR bits to determine if the mmu is being accessed in a priviledged or problem context (i still don't remember which of those happen with msr_pr == true, lol). Probably ctx->pr is a more general privileged thing than msr_pr.

Go ahead, nothing to see here :)